Doc/reference/datamodel.rst

```diff
@@ -2373,31 +2387,37 @@ Emulating container types
 -------------------------
 
 The following methods can be defined to implement container objects.  Containers
-usually are sequences (such as lists or tuples) or mappings (like dictionaries),
+usually are :term:`sequences<sequence>` (such as :class:`lists<list>` or
+:class:`tuples<tuple>`) or :term:`mappings<mapping>` (like
+:class:`dictionaries<dict>`),
 but can represent other containers as well.  The first set of methods is used
 either to emulate a sequence or to emulate a mapping; the difference is that for
 a sequence, the allowable keys should be the integers *k* for which ``0 <= k <
-N`` where *N* is the length of the sequence, or slice objects, which define a
+N`` where *N* is the length of the sequence, or :class:`slice` objects, which define a
 range of items.  It is also recommended that mappings provide the methods
-:meth:`keys`, :meth:`values`, :meth:`items`, :meth:`get`, :meth:`clear`,
-:meth:`setdefault`, :meth:`pop`, :meth:`popitem`, :meth:`!copy`, and
-:meth:`update` behaving similar to those for Python's standard dictionary
-objects.  The :mod:`collections.abc` module provides a
-:class:`~collections.abc.MutableMapping`
-abstract base class to help create those methods from a base set of
-:meth:`__getitem__`, :meth:`__setitem__`, :meth:`__delitem__`, and :meth:`keys`.
+:meth:`~dict.keys`, :meth:`~dict.values`, :meth:`~dict.items`,
```

Comment:
Not sure about this. You add a lot of links, but this sentence mentions not referred methods, but new methods. And there is already a link to dict below.

Reply:
The issue, I think, is that the documentation here says you must implement these methods, but doesn't say why, what they should do, or what they mean. That would be out of scope for this piece of documentation, but I do think readers need a clearer pointer for where they should go to find out more about how these methods work when properly implemented.

Reply:
I'll think about how I can rework this bit 🙂

Reply:
On second thoughts, you're right, the link to the documentation for `dict` is pretty obvious.